trajopt_sco/include/trajopt_sco/osqp_interface.hpp

```diff
@@ -63,9 +84,40 @@ class OSQPModel : public Model
 
 public:
   OSQPModel();
-  ~OSQPModel() override;
-  OSQPModel(const OSQPModel&) = default;
-  OSQPModel& operator=(const OSQPModel&) = default;
+  ~OSQPModel() = default;
+  OSQPModel(const OSQPModel& model) = delete;
+  OSQPModel& operator=(const OSQPModel& model)
+  {
+    osqp_settings_ = model.osqp_settings_;
+    osqp_data_ = model.osqp_data_;
+    osqp_workspace_ = std::make_unique<OSQPWorkspaceInterface>(*model.osqp_workspace_);
+
+    vars_ = model.vars_;
+    cnts_ = model.cnts_;
+    lbs_ = model.lbs_;
+    ubs_ = model.ubs_;
+    cnt_exprs_ = model.cnt_exprs_;
+    cnt_types_ = model.cnt_types_;
+    solution_ = model.solution_;
+
+    A_.reset(model.osqp_data_.A);
```

Comment:
Same issue hear. I think what you want is what is shown below. (It may be easier to just delete the copy and assignment operator and force the user to construct a new OSQPModel.) You should still be able to have a move operator.

```
osqp_data_ = OSQPData();
A_.reset(osqp_data_.A);
B_.reset(osqp_data_.B);
```